inventory-service/src/main/java/com/petclinic/inventoryservice/presentationlayer/InventoryController.java

```diff
@@ -207,6 +207,14 @@ public Mono<ResponseEntity<ProductResponseDTO>> addSupplyToInventory(@RequestBod
     public Mono<Integer> getQuantityOfProductsInInventory (@PathVariable String inventoryId) {
         return productInventoryService.getQuantityOfProductsInInventory(inventoryId);
     }
+
+    @PatchMapping("/{inventoryId}/products/{productId}/consume")
+    public Mono<ResponseEntity<ProductResponseDTO>> consumeProduct(@PathVariable String inventoryId,
+                                                                   @PathVariable String productId) {
+        return productInventoryService.consumeProduct(inventoryId, productId)
+                .map(productResponseDTO -> ResponseEntity.ok().body(productResponseDTO))
+                .defaultIfEmpty(ResponseEntity.notFound().build());
+    }
```

Comment:
The use of the PatchMapping annotation for our endpoint makes perfect sense since we are only updating partially our DB object. For this unique case this was very well executed! Good work!